packages/monaco/src/split-editor.tsx

```diff
@@ -89,6 +91,49 @@ export default function SplitEditor({
   const monacoRef = useRef<typeof import('monaco-editor')>();
   const editorRef = useRef<monacoType.editor.IStandaloneCodeEditor>();
 
+  const [isWrapperInnerFocused, setIsWrapperInnerFocused] = useState(false);
+
+  useEffect(() => {
+    const focusHandler = () => {
+      setIsWrapperInnerFocused(true);
+    };
+
+    const blurHandler = () => {
+      setIsWrapperInnerFocused(false);
+    };
+
+    wrapper.current?.addEventListener('focus', focusHandler, true);
+    wrapper.current?.addEventListener('blur', blurHandler, true);
+
+    return () => {
+      wrapper.current?.removeEventListener('focus', focusHandler, true);
+      wrapper.current?.removeEventListener('blur', blurHandler, true);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!isWrapperInnerFocused) {
+      return;
+    }
+
+    const saveHandler = (e: KeyboardEvent) => {
+      if (e.ctrlKey && e.code === 'KeyS') {
```

Comment:
cant we determine if the wrapper is in focus during this effect with its ref vs introducing a piece of state and another effect?
 
using ref.current.activeElement or hasFocus. cant remember the apis

Reply:
I will try that sure!